nexus/proposals/api.py

```diff
@@ -0,0 +1,35 @@
+from django.utils import timezone
+
+# Third Party Stuff
+from rest_framework import viewsets
+from rest_framework.decorators import action
+from rest_framework import mixins
+from rest_framework.permissions import IsAuthenticated
+
+from . import models, serializers
+from .permissions import IsCoreOrganizer, IsOwnerOrReadOnly
+
+
+class ProposalViewSet(mixins.ListModelMixin, mixins.CreateModelMixin,
+                      mixins.UpdateModelMixin, mixins.RetrieveModelMixin,
+                      mixins.DestroyModelMixin, viewsets.GenericViewSet):
+    queryset = models.Proposal.objects.all()
+    serializer_class = serializers.ProposalSerializer
+    permission_classes = (IsAuthenticated,)
+
+    @action(methods=['PATCH'], detail=True, permission_classes=(IsCoreOrganizer,))
+    def approve(self, request, pk=None):
+        data = {'status': 'accepted', 'approved_at': timezone.now()}
+        update_proposal_status(data)
+
+    @action(methods=['PATCH'], detail=True, permission_classes=(IsOwnerOrReadOnly,))
```

Comment:
Since the method this action is allowing is `PATCH` and no other read-only methods are allowed, there's no need to provide read-only permissions from `IsOwnerOrReadOnly`. Refactor the permission class and just name it `IsOwner`.